### PR TITLE
Adding Testcase for DS1189

### PR DESCRIPTION
**Pull request by anupama-pathirage**

### CODE REVIEW

File: modules/integration/tests-integration/tests/src/test/java/org/wso2/dss/integration/test/jira/issues/DS1189LeagyTimeStampModeTestCase.java [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+        };
+
+        testServerManager.setParameter("shFilename", "wso2serverLegacyMode.sh");
+
+        String testServerCarbonHome = testServerManager.startServer();
+
+        backupUserDir = System.getProperty("user.dir");
+        backupCarbonHome = System.getProperty("carbon.home");
+
+        System.out.println("****************  back up user dir   " + backupUserDir);
```
Comment (Manuri) [inline modules/integration/tests-integration/tests/src/test/java/org/wso2/dss/integration/test/jira/issues/DS1189LeagyTimeStampModeTestCase.java]: Shouldn't we be using log here? Also, if these two lines were just for testing purposes we can remove them.

File: modules/integration/tests-integration/tests/src/test/java/org/wso2/dss/integration/test/jira/issues/DS1189LeagyTimeStampModeTestCase.java [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+/**
+ * This test is to verify the fix for https://wso2.org/jira/browse/DS-1189
+ */
+public class DS1189LeagyTimeStampModeTestCase extends DSSIntegrationTest {
+
+    private static final Log log = LogFactory.getLog(DS1189LeagyTimeStampModeTestCase.class);
+
+    private final String serviceName = "TimeStampDifferenceLegacy";
+    private String backendUrl = "https://localhost:10653/services/";
```
Comment (Manuri) [inline modules/integration/tests-integration/tests/src/test/java/org/wso2/dss/integration/test/jira/issues/DS1189LeagyTimeStampModeTestCase.java]: Can't we get the back end URL via a test framework method?

File: modules/integration/tests-integration/tests/src/test/java/org/wso2/dss/integration/test/jira/issues/DS1189LeagyTimeStampModeTestCase.java [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+
+            }
+        };
+
+        testServerManager.setParameter("shFilename", "wso2serverLegacyMode.sh");
+
+        String testServerCarbonHome = testServerManager.startServer();
+
+        backupUserDir = System.getProperty("user.dir");
+        backupCarbonHome = System.getProperty("carbon.home");
```
Comment (lahirus) [inline modules/integration/tests-integration/tests/src/test/java/org/wso2/dss/integration/test/jira/issues/DS1189LeagyTimeStampModeTestCase.java]: Could we use constants here?

File: modules/integration/tests-integration/tests/src/test/java/org/wso2/dss/integration/test/jira/issues/DS1189LeagyTimeStampModeTestCase.java [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+
+        Iterator iterator = result.getChildrenWithLocalName("timeStamp");
+
+        String timeStampString = null;
+
+        while (iterator.hasNext()) {
+            OMElement timeStamp = (OMElement) iterator.next();
+            if (timeStampString == null) {
+                timeStampString = timeStamp.getChildrenWithLocalName("testTimeStamp").next().toString();
+                log.info("TimeStamp Recv:"+timeStampString);
```
Comment (lahirus) [inline modules/integration/tests-integration/tests/src/test/java/org/wso2/dss/integration/test/jira/issues/DS1189LeagyTimeStampModeTestCase.java]: Minor formatting issue. Add spaces before and after + sign. Also, the log would be clear if a space is kept as "TimeStamp Recv: ".

File: modules/integration/tests-integration/tests/src/test/java/org/wso2/dss/integration/test/jira/issues/DS1189LeagyTimeStampModeTestCase.java [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+                Assert.assertTrue(timeStampString.contains("1970-01-02T05:00:00.000+00:00"));
+            } else {
+                String tempTimeStamp = timeStamp.getChildrenWithLocalName("testTimeStamp").next().toString();
+                log.info("Timestamp Comapre:"+timeStampString+"|"+tempTimeStamp);
```
Comment (lahirus) [inline modules/integration/tests-integration/tests/src/test/java/org/wso2/dss/integration/test/jira/issues/DS1189LeagyTimeStampModeTestCase.java]: Minor formatting issue. Add spaces before and after + sign.

File: modules/integration/tests-integration/tests/src/test/java/org/wso2/dss/integration/test/jira/issues/DS1189LeagyTimeStampModeTestCase.java (hunk 3 above)
Comment (anupama-pathirage) [inline modules/integration/tests-integration/tests/src/test/java/org/wso2/dss/integration/test/jira/issues/DS1189LeagyTimeStampModeTestCase.java]: Fixed with PR: #97

File: modules/integration/tests-integration/tests/src/test/java/org/wso2/dss/integration/test/jira/issues/DS1189LeagyTimeStampModeTestCase.java (hunk 4 above)
Comment (anupama-pathirage) [inline modules/integration/tests-integration/tests/src/test/java/org/wso2/dss/integration/test/jira/issues/DS1189LeagyTimeStampModeTestCase.java]: Fixed with PR: #97

File: modules/integration/tests-integration/tests/src/test/java/org/wso2/dss/integration/test/jira/issues/DS1189LeagyTimeStampModeTestCase.java (hunk 5 above)
Comment (anupama-pathirage) [inline modules/integration/tests-integration/tests/src/test/java/org/wso2/dss/integration/test/jira/issues/DS1189LeagyTimeStampModeTestCase.java]: Fixed with PR: #97

File: modules/integration/tests-integration/tests/src/test/java/org/wso2/dss/integration/test/jira/issues/DS1189LeagyTimeStampModeTestCase.java (hunk 1 above)
Comment (chrishantha) [inline modules/integration/tests-integration/tests/src/test/java/org/wso2/dss/integration/test/jira/issues/DS1189LeagyTimeStampModeTestCase.java]: Yes. Remove the System.out.println statements

Comment (anupama-pathirage) [inline modules/integration/tests-integration/tests/src/test/java/org/wso2/dss/integration/test/jira/issues/DS1189LeagyTimeStampModeTestCase.java]: Fixed with PR : #97

**Pull request merged**